### PR TITLE
Function/is correlation

### PR DESCRIPTION
**Pull request by atrayees**
wrote the function ` is_correlation_matrix() ` inside volesti/include, and further updates the example sampler which calls this very function.

### CODE REVIEW

File: include/matrix_operations/EigenvaluesProblems.h [hunk 1]
```diff
@@ -439,6 +439,26 @@ class EigenvaluesProblems<NT, Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic>, E
...
 
+    /// Check if a matrix is indeed a correlation matrix
+    /// return true if input matrix is found to be a correlation matrix
+    /// |param[in] matrix
+    bool is_correlation_matrix(const MT& matrix){
+
+        const double tol = 1e-8;
+    
+        //check if all the diagonal elements are ones
+        for(int i=0 ; i<matrix.rows() ; i++){
```
Comment (vfisikop) [inline include/matrix_operations/EigenvaluesProblems.h]: please leave one empty space between `for` and `(`, similar to `if` etc

File: include/matrix_operations/EigenvaluesProblems.h [hunk 2]
```diff
@@ -439,6 +439,26 @@ class EigenvaluesProblems<NT, Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic>, E
         return false;
     }
 
+    /// Check if a matrix is indeed a correlation matrix
+    /// return true if input matrix is found to be a correlation matrix
+    /// |param[in] matrix
+    bool is_correlation_matrix(const MT& matrix){
+
+        const double tol = 1e-8;
```
Comment (vfisikop) [inline include/matrix_operations/EigenvaluesProblems.h]: This could be a function parameter with default value.

**Review by vfisikop — APPROVED** [review]
Thanks!

**Review by TolisChal — APPROVED** [review]

File: include/matrix_operations/EigenvaluesProblems.h (hunk 1 above)
Comment (atrayees) [reply include/matrix_operations/EigenvaluesProblems.h]: done!

File: include/matrix_operations/EigenvaluesProblems.h (hunk 2 above)
Comment (atrayees) [reply include/matrix_operations/EigenvaluesProblems.h]: done

**Pull request merged**